### PR TITLE
FEATURE: add the set user status button to the new user menu

### PR DESCRIPTION
**Pull request by AndrewPrigorshnev**
Screenshot 2022-08-24 at 22 11 32

Screenshot 2022-08-24 at 22 11 22

### CODE REVIEW

File: app/assets/javascripts/discourse/tests/acceptance/user-status-test.js [hunk 1]
```diff
@@ -340,3 +340,83 @@ acceptance("User Status", function (needs) {
...
+  });
+
+  test("shows the user status button on the menu when enabled in settings", async function (assert) {
+    this.siteSettings.enable_user_status = true;
+
+    await visit("/");
+    await click(".header-dropdown-toggle.current-user");
+    await click("#user-menu-button-profile");
+
+    assert.ok(exists("li.set-user-status"), "shows the button");
```
Comment (OsamaSayegh) [inline app/assets/javascripts/discourse/tests/acceptance/user-status-test.js:387]:
```suggestion
    assert.ok(exists("li.set-user-status .btn"), "shows the button");
```

File: app/assets/javascripts/discourse/tests/acceptance/user-status-test.js [hunk 2]
```diff
@@ -340,3 +340,83 @@ acceptance("User Status", function (needs) {
...
+    updateCurrentUser({
+      status: { description: userStatus, emoji: userStatusEmoji },
+    });
+
+    await visit("/");
+    await click(".header-dropdown-toggle.current-user");
+    await click("#user-menu-button-profile");
+
+    assert.equal(
+      query("li.set-user-status .item-label").innerText,
```
Comment (OsamaSayegh) [inline app/assets/javascripts/discourse/tests/acceptance/user-status-test.js:405]:
```suggestion
      query("li.set-user-status .item-label").textContent.trim(),
```

I recently learned that `innerText` of an element can change depending on the CSS styles applied to the element, so it's probably a little more future-proof if we use `textContent`.

File: app/assets/stylesheets/common/base/menu-panel.scss [hunk 3]
```diff
@@ -208,6 +208,20 @@
...
+        padding-top: 0;
+      }
+
+      .relative-date {
+        text-align: left;
+        font-size: var(--font-down-3);
+        padding-top: 0.25em;
+        margin-left: 0.75em;
+        color: var(--primary-medium);
+      }
```
Comment (OsamaSayegh) [inline app/assets/stylesheets/common/base/menu-panel.scss:222]: We do have some CSS for `.relative-date` at line 236 for the Do Not Disturb button, maybe we can reuse some of that CSS?

**Review by OsamaSayegh — APPROVED** [review]
Some minor comments but otherwise it looks great!

File: app/assets/stylesheets/common/base/menu-panel.scss (hunk 3 above)
Comment (AndrewPrigorshnev) [reply app/assets/stylesheets/common/base/menu-panel.scss]: We can for sure, I did that.

**Comment by AndrewPrigorshnev** [comment]
@OsamaSayegh thank you for review!

**Pull request merged**